src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -1853,9 +1855,34 @@ static IEnumerable<Decorator> GetBicepTemplateDecorators()
                 yield return new(decorator, (_, _) => true);
             }
 
+            //foreach (var decorator in GetBicepTemplateDecorators())
+            //{
+            //    yield return new(decorator, (_, sfk) => sfk == BicepSourceFileKind.BicepFile);
+            //}
             foreach (var decorator in GetBicepTemplateDecorators())
             {
-                yield return new(decorator, (_, sfk) => sfk == BicepSourceFileKind.BicepFile);
+                if (decorator.Overload.Name == LanguageConstants.ParameterSecurePropertyName)
+                {
+                    yield return new(decorator, (features, sfk) =>
+                    {
+                        var isValid = sfk == BicepSourceFileKind.BicepFile;
+
+                        if (features.SecureOutputsEnabled)
+                        {
+                            isValid &= (decorator.Overload.Flags & FunctionFlags.ParameterOutputOrTypeDecorator) == decorator.Overload.Flags;
+                        }
+                        else
+                        {
+                            isValid &= (decorator.Overload.Flags & FunctionFlags.ParameterOrTypeDecorator) == decorator.Overload.Flags;
+                        }
+                        return isValid;
+                    });
+                }
```

Comment:
I don't think the visibility delegate is the right place to put this logic -- we still want `@secure()` to be visible if the feature is not enabled, but it needs to have a different flag. I would recommend removing the `static readonly` field containing the decorators array and instead pass an `IFeatureProvider` to `GetSystemDecorators` and call it [from within the `Create` method](https://github.com/Azure/bicep/pull/14661/commits/31aeec28197737e3f80198e3adc9a3de70e9e640#diff-8b52fb453e20d73d655285b4895fac59e7816d816043e92ac4a313e52056c551R1988)